src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +65,8 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    public boolean isMutating(){
```

Comment:
Would it be better for this method to be abstract?
By making it abstract, you force child classes to override this method, i.e. making them declare whether they mutate the data or not.
Also, you should have a header comment here to describe isMutating, so that other developers making child classes will know how to override it.